### PR TITLE
Add Mirage export functionality

### PR DESCRIPTION
**Pull request by noughtmare**
Mirage is a tool that can create visualizations of attribute grammar (see https://github.com/Helium4Haskell/mirage). It is intended as a successor of Visage. For this I implemented an additional way to export information. I have tried to keep the changes as small and simple as possible on the side of UUAGC. I decided not to use the existing Visage export functionality because it uses outdated ATerms as format and it didn't export all the information that I needed. I think the Visage code can be removed, but that is not part of this pull request.

### CODE REVIEW

File: uuagc/trunk/src/JSON.hs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+module JSON where
```
Comment (jbransen) [inline uuagc/trunk/src/JSON.hs]: I'd prefer to remove this module and depend on [`aeson`](http://hackage.haskell.org/package/aeson) instead, which is the standard for JSON encoding of Haskell data types.

File: .gitignore [hunk 2]
```diff
@@ -5,3 +5,4 @@
 uuagc/trunk/dist/*
 dist-newstyle/
 cabal.project.local
+cabal.project.local~
```
Comment (jbransen) [inline .gitignore:8]: This looks like an emacs backup file, probably better to add `*~` here

**Review by jbransen — CHANGES_REQUESTED** [review]
Thanks for the MR! If you could remove the custom JSON class and use `aeson` instead that would be great.

File: uuagc/trunk/src/JSON.hs (hunk 1 above)
Comment (noughtmare) [reply uuagc/trunk/src/JSON.hs]: Ah, I chose to do it this way to add minimal dependencies to UUAGC, but if you're fine with Aeson then that also works.

File: .gitignore (hunk 2 above)
Comment (noughtmare) [reply .gitignore:8]: I think it is actually generated by running `cabal configure` (I use vim myself). If the `cabal.project.local` file already exists, it saves the old file with that suffix.

**Comment by noughtmare** [comment]
I have now updated it to use Aeson (and bytestring).

File: .gitignore (hunk 2 above)
Comment (noughtmare) [reply .gitignore:8]: The "official" haskell gitignore also has `cabal.project.local~` listed: https://github.com/github/gitignore/blob/master/Haskell.gitignore. So I think leaving it like this is fine.

**Pull request merged**